Exception when selecting an item without hash

The following exception is thrown whenever an item without a hash value is selected.

```
2022-01-25 16:51:45	[INFO]	[indexer.desktop.FileProcessor]			Opening XXXXX.ufdr/_DecodedData/InstantMessage/InstantMessage_YYYYY
java.lang.RuntimeException: iped3.exception.QueryNodeException: INVALID_SYNTAX_CANNOT_PARSE: Syntax Error, cannot parse linkedItems:() sharedHashes:():  
	at dpf.sp.gpinf.indexer.search.IPEDSearcher.setQuery(IPEDSearcher.java:91)
	at dpf.sp.gpinf.indexer.search.IPEDSearcher.<init>(IPEDSearcher.java:71)
	at dpf.sp.gpinf.indexer.desktop.ReferencesTableModel.listReferencingItems(ReferencesTableModel.java:200)
	at dpf.sp.gpinf.indexer.desktop.FileProcessor.process(FileProcessor.java:175)
	at dpf.sp.gpinf.indexer.desktop.FileProcessor.doInBackground(FileProcessor.java:117)
	at dpf.sp.gpinf.indexer.desktop.FileProcessor.doInBackground(FileProcessor.java:47)
	at java.desktop/javax.swing.SwingWorker$1.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.desktop/javax.swing.SwingWorker.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: iped3.exception.QueryNodeException: INVALID_SYNTAX_CANNOT_PARSE: Syntax Error, cannot parse linkedItems:() sharedHashes:():  
	at dpf.sp.gpinf.indexer.search.QueryBuilder.getQuery(QueryBuilder.java:292)
	at dpf.sp.gpinf.indexer.search.QueryBuilder.getQuery(QueryBuilder.java:249)
	at dpf.sp.gpinf.indexer.search.IPEDSearcher.setQuery(IPEDSearcher.java:88)
	... 11 more
Caused by: INVALID_SYNTAX_CANNOT_PARSE: Syntax Error, cannot parse linkedItems:() sharedHashes:():  
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.generateParseException(StandardSyntaxParser.java:880)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.jj_consume_token(StandardSyntaxParser.java:748)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.Clause(StandardSyntaxParser.java:277)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.ModClause(StandardSyntaxParser.java:215)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.ConjQuery(StandardSyntaxParser.java:152)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.DisjQuery(StandardSyntaxParser.java:129)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.Query(StandardSyntaxParser.java:91)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.GroupingExpr(StandardSyntaxParser.java:294)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.Clause(StandardSyntaxParser.java:265)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.ModClause(StandardSyntaxParser.java:215)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.ConjQuery(StandardSyntaxParser.java:152)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.DisjQuery(StandardSyntaxParser.java:129)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.Query(StandardSyntaxParser.java:91)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.TopLevelQuery(StandardSyntaxParser.java:81)
	at org.apache.lucene.queryparser.flexible.standard.parser.StandardSyntaxParser.parse(StandardSyntaxParser.java:67)
	at org.apache.lucene.queryparser.flexible.core.QueryParserHelper.parse(QueryParserHelper.java:214)
	at org.apache.lucene.queryparser.flexible.standard.StandardQueryParser.parse(StandardQueryParser.java:148)
	at dpf.sp.gpinf.indexer.search.QueryBuilder.getQuery(QueryBuilder.java:285)
	... 13 more
```

It seems to be harmless, as it only prints this stack trace in IPED-SearchApp.log.
I detected it while using the latest code (master), but after testing in 3.18.12, it seems that it also happens in 3.18.x versions.

--- a/iped-app/src/main/java/dpf/sp/gpinf/indexer/desktop/ReferencesTableModel.java
+++ b/iped-app/src/main/java/dpf/sp/gpinf/indexer/desktop/ReferencesTableModel.java
@@ -193,31 +193,37 @@ public class ReferencesTableModel extends AbstractTableModel
         String edonkey = doc.get(HashTask.HASH.EDONKEY.toString());
         String hashes = Arrays.asList(md5, sha1, sha256, edonkey).stream().filter(a -> a != null)
                 .collect(Collectors.joining(" "));
-        String textQuery = ExtraProperties.LINKED_ITEMS + ":(" + hashes + ") ";
-        textQuery += ExtraProperties.SHARED_HASHES + ":(" + hashes + ")";
-
-        try {
-            IPEDSearcher task = new IPEDSearcher(App.get().appCase, textQuery);
-            results = task.luceneSearch();
-
-            final int length = results.getLength();
-
-            if (length > 0) {
-                SwingUtilities.invokeLater(new Runnable() {
-                    @Override
-                    public void run() {
-                        App.get().referencesDock.setTitleText(length + Messages.getString("ReferencesTab.Title"));
-                    }
-                });
-                refDoc = doc;
-            } else {
-                refDoc = null;
-            }
-
-        } catch (Exception e) {
+        
+        if (hashes.isEmpty()) {
             results = new LuceneSearchResult(0);
             refDoc = null;
-            e.printStackTrace();
+        } else {
+            String textQuery = ExtraProperties.LINKED_ITEMS + ":(" + hashes + ") ";
+            textQuery += ExtraProperties.SHARED_HASHES + ":(" + hashes + ")";
+    
+            try {
+                IPEDSearcher task = new IPEDSearcher(App.get().appCase, textQuery);
+                results = task.luceneSearch();
+    
+                final int length = results.getLength();
+    
+                if (length > 0) {
+                    SwingUtilities.invokeLater(new Runnable() {
+                        @Override
+                        public void run() {
+                            App.get().referencesDock.setTitleText(length + Messages.getString("ReferencesTab.Title"));
+                        }
+                    });
+                    refDoc = doc;
+                } else {
+                    refDoc = null;
+                }
+    
+            } catch (Exception e) {
+                results = new LuceneSearchResult(0);
+                refDoc = null;
+                e.printStackTrace();
+            }
         }
 
         fireTableDataChanged();

--- a/iped-app/src/main/java/dpf/sp/gpinf/indexer/desktop/ReferencesTableModel.java
+++ b/iped-app/src/main/java/dpf/sp/gpinf/indexer/desktop/ReferencesTableModel.java
@@ -18,7 +18,6 @@
  */
 package dpf.sp.gpinf.indexer.desktop;
 
-import java.awt.Rectangle;
 import java.awt.event.MouseEvent;
 import java.awt.event.MouseListener;
 import java.util.Arrays;

